lib/readline.js

```diff
@@ -481,6 +482,42 @@ Interface.prototype._insertString = function(c) {
     // A hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+  // Emit current line for generating preview
+  this.emit('buffer', this.line);
+};
+
+// Append eager eval result to the
+// Current line
+Interface.prototype._appendPreview = function(result) {
+  this.previewResult = `\u001b[90m // ${result}\u001b[39m`;
+  const columns = this.output.columns;
+
+  // Cursor to left edge.
+  cursorTo(this.output, 0);
+  clearScreenDown(this.output);
+
+  if (columns) {
+    const colorDepth = this.output.getColorDepth();
+    const s = colorDepth > 256 ?
+      `${this._prompt}${this.line}${this.previewResult}` :
+      `${this._prompt}${this.line}${result}`;
+    this.output.write(s.length < columns ?
```

Comment:
The length check includes the colors but that should ideally not be the case. Colors are not visible characters and should as such not be calculated towards the total length.